File: c/estrutura_condicional/glicose.c
/* Problema "glicose"
 Fazer um programa para ler a quantidade de glicose
 no sangue de uma pessoa e depois mostrar na tela a
 classificação desta glicose de acordo com a tabela de
 referência abaixo:

 |----------------------------------------------|
 | Classificação  | Glicose                     |
 |----------------------------------------------|
 | Normal         | Até 100 mg/dl               |
 | Elevado        | Maior que 100 até 140 mg/dl |
 | Diabetes       | Maior de 140 mg/dl          |
 |----------------------------------------------|
*/

#include <stdio.h>
#include <string.h>

int main()
{
  const double NORMAL_LEVEL = 100.0;
  const double HIGH_LEVEL = 140.0;
  double glucoseLevel;
  char classification[20] = "diabetes";

  printf("Digite a medida da glicose: ");
  scanf("%lf", &glucoseLevel);

  if (glucoseLevel <= NORMAL_LEVEL)
  {
    strcpy(classification, "normal");
  }
  else if (glucoseLevel <= HIGH_LEVEL)
  {
    strcpy(classification, "elevado");
  }

  printf("Classificação = %s\n", classification);

  return 0;
}

File: csharp/Source/Estruturas/Condicional/Glicose.cs
namespace Source.Estruturas.Condicional;

/**
 * Problema "glicose"
 * Fazer um programa para ler a quantidade de glicose
 * no sangue de uma pessoa e depois mostrar na tela a
 * classificação desta glicose de acordo com a tabela de
 * referência abaixo:
 * |----------------------------------------------|
 * | Classificação  | Glicose                     |
 * |----------------------------------------------|
 * | Normal         | Até 100 mg/dl               |
 * | Elevado        | Maior que 100 até 140 mg/dl |
 * | Diabetes       | Maior de 140 mg/dl          |
 * |----------------------------------------------|.
 */

internal static class Glicose
{

    private static readonly double NORMAL_LEVEL = 100.0;

    private static readonly double HIGH_LEVEL = 140.0;

    public static void Execute()
    {
        Console.Write("Digite a medida da glicose: ");
        double glucoseLevel = double.Parse(Console.ReadLine()!);

        string classification = "diabetes";
        if (glucoseLevel <= NORMAL_LEVEL)
        {
            classification = "normal";
        }
        else if (glucoseLevel <= HIGH_LEVEL)
        {
            classification = "elevado";
        }

        Console.WriteLine("Classificação = {0}", classification);
    }
}